### PR TITLE
User auth

### PR DESCRIPTION
**Pull request by samuelmoiwa**
## User auth

use the device gem to create user auth

- [ ] sign up
- [ ] sign in
- [ ] forgetting password
- [ ] reset password

### CODE REVIEW

File: app/views/devise/registrations/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+<%= form_for(resource, as: resource_name, url: registration_path(resource_name)) do |f| %>
+  <%= render "devise/shared/error_messages", resource: resource %>
+
+  <div class="field">
+    <%= f.label :name %><br />
+    <%= f.text_field :name %>
+  </div>
+
+  <div class="field">
```
Comment (Irirwanirira) [inline app/views/devise/registrations/new.html.erb:11]: - [ ] As mentioned in the[ project](https://github.com/microverseinc/curriculum-rails/blob/main/recipe-app/buisness_requirements.md), please use only users name, to meet the project requirements.

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -11,6 +11,8 @@
   </head>
 
   <body>
```
Comment (Irirwanirira) [inline app/views/layouts/application.html.erb:13]: - [ ] Kindly, please add these lines of codes to the layout to allow easy log out after the user has logged in.
log_out

File: db/migrate/20230214192802_add_name_to_users.rb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+class AddNameToUsers < ActiveRecord::Migration[7.0]
+  def change
+    add_column :users, :name, :string
+    add_column :users, :surname, :string
```
Comment (Irirwanirira) [inline db/migrate/20230214192802_add_name_to_users.rb]: - [ ] Kindly, as mentioned in the project requirement, only use the user's name.

**Review by Irirwanirira — CHANGES_REQUESTED** [review]
## Status: Required Changes ♻️ ♻️ ♻️

Hi @samuelmoiwa , I'm @Irirwanirira  your reviewer.

Good job so far in implementing the following changes!
But there are some issues that you still need to work on to go to the next project. but you are almost there!


## Highlights

✅ You used Gitflow correctly 👍🏽
✅ Linter checks passed 👍🏽
✅ Setup devise for authentication 👍🏽

## Required Changes ♻️

_Check the comments under the review._

- [ ] Kindly modify the style of the views generated by devise to match the [wireframe](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/sneak_peek.md) given for this project. At the moment the login, signup page, post creation, post list page, and post details page are not styled according to te wireframe. Please add some styles to match that wireframe design 👍🏽
- [ ] Kindly solve the conflicts between your branch and the default branch, to allow a merge.
- [ ] Kindly  update the application controller at app/controllers/application_controller.rb to always require a user to be logged in for all actions:
exception

- [ ] Kindly use this full configuration to set up your application controller at app/controllers/application_controller.rb,to set database params, and when the devise activated.
protected

- [ ] As you mentioned in the routes.rb, you specified "users#index" as the root route, yet User is not defined in the controller not even in the view to be displayed after the user  signs in. Please fix that issue.



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Comment by Kaweesi-Matia** [comment]
Hi @Irirwanirira 
Your implementation for the login of the APP is complete! There is nothing else to say other than... it's time to merge it :shipit:

Congratulations! 🎉

**Pull request merged**